### ISSUE TITLE
Ling stasis can rez you inside blob zombies

### ISSUE THREAD

**Issue by TemporalOroboros** [tgstation/tgstation#49543]
##Round ID:
Issue reported from  Round ID: 130873 (/tg/Station Bagil [ENGLISH] [US-WEST] [100% FREE LAG])

Reporting client version: 513.1507

##Reproduction:

1. Be a ling
2. Have a blob spawn
3. Fight the blob
4. Use Regenerative Stasis
5. Die while in stasis
6. Have your corpse possessed by a blob spore
7. Revive from stasis

You are now riding around in the blob zombie created from your corpse. You can't control it and it will act as a normal blob zombie until it dies, at which point instead of releasing your corpse it will release you, alive and well! In this state, since you are technically 'alive', you regenerate chemicals and can use your abilities. (The only one that I tested was the transformation ability which didn't have any visible effect until I was released.) If you happened to buy and activate the armblade before you died you can attack the blob zombie to kill it and release yourself pretty much at will, you can't attack anything else though.

**Comment by Dennok** [comment]
Army of blob zombied lings.

**Comment by HomocidlePancakes** [comment]
That’s intended. Same thing happens if you get eaten by morph or legion. Use your armblade to break out.

**Comment by XDTM** [comment]
"Having a workaround" is not the same as "intended"

**Comment by Fikou** [comment]
feature

**Comment by HomocidlePancakes** [comment]
What do you mean “Having a workaround”? Cling cutting his way out of a monster that ate/absorbed him makes perfect sense to me. The only way this can be a bug is if you could break out only using an armblade.

**Comment by XDTM** [comment]
a blob zombie is a corpse with a spore on its head, it does not "contain" a real changeling that needs to hack its way out

**Comment by TemporalOroboros** [comment]
@XDTM Given the code for the blob spores I'm pretty sure that the blob zombie _literally_ contains the ling.
```
/mob/living/simple_animal/hostile/blob/blobspore/proc/Zombify(mob/living/carbon/human/H)
	is_zombie = 1
        ...
        H.update_hair()
	H.forceMove(src)
	oldguy = H
        ...
```

**Comment by ATH1909** [comment]
Yeah, but fluff-wise, the blob spore is just covering the ling's head, not their entire body.

**Comment by XDTM** [comment]
Yes, the code is that way because it makes handling the AI easier than trying to manipulate the normal human body; it does result in _unintended_ inconsistencies in the case the corpse starts being alive again.

**Comment by nemvar** [comment]
Listen to a signal fired from revive that kill the blob spore (or makes it detach).
That's the most fancy solution I can think off.

**Comment by Mooshimi** [comment]
Saw a ling revive and drop out of a dragon the other week.

**Comment by timothymtorres** [comment]
List of closed issues where this was happening:

- #31258
- #52432
- #27757
- #29213
- #32069

Someone made #31624 in an attempt to fix the blob zombie thing but it never got merged.

Laughter/Slaughter demons have custom code to handle ling revival when they are devoured here:

https://github.com/tgstation/tgstation/blob/45d2cdb3187e9b1b5a03364907e67916529735f9/code/modules/spells/spell_types/jaunt/bloodcrawl.dm#L278-L296

The changeling revive code is here:

https://github.com/tgstation/tgstation/blob/04f43d6d45825b1095c868e79c7b6fb17f59e06f/code/modules/antagonists/changeling/powers/fakedeath.dm#L30-L47

I think someone just needs to add a simple: `isliving(user.loc)` to check if the ling has been devoured by a mob and then teleport them out of the location with a message informing both mobs.

**Comment by Sealed101** [comment]
i think there's merit in removing ling revival when one is blob spored to prevent any dumbfucks from trying to abuse it (i remember you gum_disease) but otherwise it's a feature

**Comment by Mooshimi** [comment]
> feature

do u still stand by this since you're now a maintainer
